### ISSUE TITLE
Misleading warning "the --global flag is deprecated" when using sandbox

### ISSUE THREAD

**Issue by edsko** [haskell/cabal#3933]
Perhaps this is a non-issue if we're going to deprecate sandboxes altogether, but right now doing a "traditional" install of a package into a sandbox yields a warning such as

```
# cabal install async
Warning: the --global flag is deprecated -- it is generally considered a bad
idea to install packages into the global store
Resolving dependencies...
Notice: installing into a sandbox located at
/home/edsko/wt/util/ghc-events-analyze/examples/blogpost/.cabal-sandbox
Configuring stm-2.4.4.1...
Building stm-2.4.4.1...
Installed stm-2.4.4.1
Configuring async-2.1.0...
Building async-2.1.0...
Installed async-2.1.0
```

This is misleading because of course I'm not actually installing anything into the global package store here.

**Comment by 23Skidoo** [comment]
I've also seen this. May be as easy as making sandboxes use `--user` instead of `--global`, though I vaguely remember that there was some issue with `--user` that made me change sandboxes to use the latter.

**Comment by ezyang** [comment]
I updated the milestone because I don't think this would be a blocker for 2.0.

**Comment by edsko** [comment]
Maybe it's just the wrong way around?

```
# cabal install ./ghcjs.cabal 
Resolving dependencies...
Configuring base16-bytestring-0.1.1.6...
Configuring byteable-0.1.1...
Configuring base-orphans-0.5.4...
....
```

(no warning), but

```
# cabal sandbox init
Writing a default package environment file to
/opt/ghcjs/HEAD-8.0/cabal.sandbox.config
Creating a new sandbox at /opt/ghcjs/HEAD-8.0/.cabal-sandbox

# cabal install ./ghcjs.cabal 
Warning: the --global flag is deprecated -- it is generally considered a bad
idea to install packages into the global store
Resolving dependencies...
Notice: installing into a sandbox located at
/opt/ghcjs/HEAD-8.0/.cabal-sandbox
Configuring ansi-terminal-0.6.2.3...
Configuring base-orphans-0.5.4...
...
```

though possibly there is something more subtle going on. Anyway, just thought I'd mention this.

**Comment by 23Skidoo** [comment]
Will try to look into this today.

**Comment by edsko** [comment]
Well it's not a priority :) I'm just recording issues as I notice them :)

**Comment by 23Skidoo** [comment]
Better sooner than later.

**Comment by rwbarton** [comment]
I also noticed this erroneous warning about `--global` while installing into a sandbox, with
```
cabal-install version 1.25.0.0
compiled using version 1.25.0.0 of the Cabal library 
```

**Comment by int-e** [comment]
The behavior was introduced in 9d59e40e9d which checks `configUserInstall` to check for the `--global` flag. However, `cabal.sandbox.config` files typically (always?) contain a line saying `user-install: false` and custom "global" installation paths, so that check is overzealous.

**Comment by 23Skidoo** [comment]
Fixed by 2ca7450ffd87a731ffade06f15567c89091a89e9.
